sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample09_MockingEventHubPublisher.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Producer;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    [TestFixture]
+    public class Sample09_MockingEventHubPublisher
+    {
+        /// <summary>
+        ///     Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        [Test]
+        public async Task EventBatch()
+        {
+            #region Snippet:EventHubs_Sample09_EventBatchMock
+
+            var size = 1024;
+            var store = new List<EventData> { new EventData(Array.Empty<byte>()), new EventData(Array.Empty<byte>()) };
+            var mockResponse = new Mock<Response>();
+            Response<EventDataBatch> response = Response.FromValue(EventHubsModelFactory.EventDataBatch(size, store), mockResponse.Object);
+
+            var mockEventHubProducerClient = new Mock<EventHubProducerClient>();
+
+            mockEventHubProducerClient.Setup(c => c.CreateBatchAsync(It.IsAny<CancellationToken>())).ReturnsAsync(response);
+
+            var producer = mockEventHubProducerClient.Object;
+
+            try
+            {
+                using var eventBatch = await producer.CreateBatchAsync();
+
+                var eventBody = new BinaryData("This is an event body");
+                var eventData = new EventData(eventBody);
+
+                if (!eventBatch.TryAdd(eventData))
```

Comment:
There's a [`tryAddCallback`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsModelFactory.cs#L126) parameter that let's the result of `TryAdd` be mocked; it would probably be a good idea to show this as well.